pkg/mtadapter/githubsource.go

```diff
@@ -20,51 +20,57 @@ import (
 	"context"
 	"fmt"
 
-	"go.uber.org/zap"
-	"k8s.io/client-go/kubernetes"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
 
+	corev1 "k8s.io/api/core/v1"
+	clientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"knative.dev/pkg/controller"
 	"knative.dev/pkg/logging"
-	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/reconciler"
 
 	"knative.dev/eventing-github/pkg/apis/sources/v1alpha1"
 	githubsourcereconciler "knative.dev/eventing-github/pkg/client/injection/reconciler/sources/v1alpha1/githubsource"
 	"knative.dev/eventing-github/pkg/common"
+	"knative.dev/eventing-github/pkg/mtadapter/router"
 )
 
-// Reconciler updates the internal Adapter cache GitHubSources
+// Reconciler manages HTTP routes based on the GitHubSource objects in the
+// controller's cache.
 type Reconciler struct {
-	kubeClientSet kubernetes.Interface
-	router        *Router
+	secrGetter clientcorev1.SecretsGetter
+	ceClient   cloudevents.Client
+	router     *router.Router
 }
 
-// Check that our Reconciler implements ReconcileKind.
+// Check that Reconciler implements reconciler.Interface.
 var _ githubsourcereconciler.Interface = (*Reconciler)(nil)
 
-func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1alpha1.GitHubSource) pkgreconciler.Event {
-	if !source.Status.IsReady() {
-		return fmt.Errorf("GitHubSource is not ready. Cannot configure the adapter")
+// ReconcileKind implements reconciler.Interface.
+func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1alpha1.GitHubSource) reconciler.Event {
```

Comment:
If you feel like it, you can look at the pattern followed by mtping and [mtkafkasource](https://github.com/knative-sandbox/eventing-kafka/pull/186/files#diff-a5c02b4c4b7add5059e10bda214ec84341aecde9fb7eb7f809c118a757cdb4ccR46). 

Soon (hopefully) the idea is to get rid of the data-plane controller and to leverage Slinky's work on data-plane <-> control-plane direct communication. Moving the Reconcile logic over to the adapter might help.

Reply:
Thanks for the references!